### DISCUSSION TITLE
openpgp.decrypt outputs serialized decimal bytes, instead of the raw bytes.

### DISCUSSION THREAD

**Discussion by Mellywins** [openpgpjs/openpgpjs#1471] [Q&A] [answered]
- OpenPGP.js version: 4.4.18
- Affected platform (Browser or Node.js version): Node 14

When piping the decrypt function into a ` createWriteStream ` , with explicit mention of ` format: binary ` . I'm getting the textual representation of the bytes in decimals.
So i'm using openpgp to encrypt a tar file with this bit of code:
```js
 const pubKey = await opengpg.readKey({armoredKey: publicKey});
  const sourceStream = createReadStream(sourcePath);
  const encrypted = await opengpg.encrypt({
    message: await opengpg.createMessage({text: sourceStream}),
    encryptionKeys: pubKey,
    format: 'binary',
  });
  encrypted
    .pipe(createWriteStream(outputPath + '.enc'))
``` 
This gives me the desired file format.
However when decrypting with this bit of code:
```js
  opengpg
    .readMessage({
      binaryMessage: sourceStream,
    })
    .then(async (M) => {
      (
        await opengpg.decrypt({
          message: M,
          decryptionKeys: pKey,
          format: 'binary',
        })
      ).data.pipe(outputStream);
``` 
I get the output in the following format:
```
31,139,8,0,0,0,0,0,0,3,237,83,201,110,219,48,16,245,167,76,238,130,43,69,178,117,44,10,244,216,83,209,31,24,147,99,123,0,110,225,98,228,243,59,164,171,68,62,228,212,4,41,10,191,131,73,141,104,206,91,70,202,219,16,41,37,246,238,203,230,131,208,247,253,60,207,32,235,48,239,250,245,186,0,54,195,52,204,143,130,105,24,55,253,48,238,247,123,216,236,62,138,208,26,37,101,140,66,229,111,239,89,132,44,235,....
```

**Comment by Mellywins** [comment]
After discussing this on gitter with @twiss, he mentioned the TypedArray#toString. Which in fact prints typed Arrays in that format: `UInt8Array [10,12,9,8] .toString() -> 10,12,9,8 ` . I don't see where the call for the toString() method is inferred from.

**Comment by twiss** [comment, marked as answer]
Probably it comes from this code, when encrypting:

> ```js
>     message: await opengpg.createMessage({text: sourceStream}),
> ```

You're saying here that the input stream is text, so it gets treated as a string. That's where the `toString` comes from.

Try

```js
    message: await opengpg.createMessage({binary: sourceStream}),
```

**Reply by Mellywins** [reply]
This was it... Finally!
This problem have taken me alot more time than it should have.. :(
